includes/Autoloader.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * AltaPay module for WooCommerce
+ *
+ * Copyright © 2020 AltaPay. All rights reserved.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+spl_autoload_register( 'altapay_namespace_autoload' );
+
+/**
+ * Dynamically loads the class attempting to be instantiated elsewhere in the
+ * plugin by looking at the $class_name parameter being passed as an argument.
+ *
+ * The argument should be in the form: TutsPlus_Namespace_Demo\Namespace. The
+ * function will then break the fully-qualified class name into its pieces and
+ * will then build a file to the path based on the namespace.
+ *
+ * The namespaces in this plugin map to the paths in the directory structure.
+ *
+ * @param string $class_name The fully-qualified name of the class to load.
+ */
+function altapay_namespace_autoload( $class_name ) {
+
+	if ( false === strpos( $class_name, 'Altapay' ) ) {
+		return;
+	}
+
+	// Split the class name into an array to read the namespace and class.
+	$file_parts = explode( '\\', $class_name );
+
+	// Do a reverse loop through $file_parts to build the path to the file.
+	$namespace = '';
+	for ( $i = count( $file_parts ) - 1; $i > 0; $i-- ) {
+		$current = $file_parts[ $i ];
+		$current = str_ireplace( '_', '-', $current );
+
+		// If we're at the first entry, then we're at the filename.
+		if ( count( $file_parts ) - 1 === $i ) {
+
+			/*
+			 * If 'interface' is contained in the parts of the file name, then
+			 * define the $file_name differently so that it's properly loaded.
+			 * Otherwise, just set the $file_name equal to that of the class
+			 * filename structure.
+			 */
+			if ( strpos( strtolower( $file_parts[ count( $file_parts ) - 1 ] ), 'interface' ) ) {
+
+				// Grab the name of the interface from its qualified name.
+				$interface_name = explode( '_', $file_parts[ count( $file_parts ) - 1 ] );
+				$interface_name = $interface_name[0];
+
+				$file_name = "interface-$interface_name.php";
+
+			} else {
+				$file_name = $current . '.php';
+			}
+		} else {
+			$namespace = '/' . $current . $namespace;
+		}
+	}
+
+	// Now build a path to the file using mapping to the file location.
+	$filepath  = trailingslashit( dirname( dirname( __FILE__ ) ) . strtolower( $namespace ) );
```

Comment:
```suggestion
	$filepath  = trailingslashit( dirname( __DIR__ ) . strtolower( $namespace ) );
```
Also if you need to move up several dirs then use the secound parameter instead of multiple calls